### ISSUE TITLE
lint empty ranges

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#11975]
### What it does

Lint on things like `3..3` which is know to be empty, or something like `x..x` which more likely is a mistake.

I'm not sure if we have anything like `Range::empty` to suggest so this would probably be a `suspicious` lint in case there is not "official" way to construct an empty range.

### Example

```rust
for i in (x..x) {
  somecode();
}
```

**Comment by ericwu2003** [comment]
There is a lint called `reversed_empty_ranges` which will lint on an expression like `for _ in 3..3` but does not lint on something like `for _ in x..x`.

I think it would be a good idea to lint on expressions such as `for _ in x..x`, since this is also likely a mistake. I'd be happy to implement this if others agree that it's a good addition to the lint.

Could someone with more experience explain? Why does the `reversed_empty_ranges` lint only trigger for empty ranges (`x..y` where `x == y`) when the range is the condition of a for loop?

**Comment by GuillaumeGomez** [comment]
I think it's closer to exception rather than what's expected in such cases. I think clippy should lint on such a case and people needing this can disable the lint.
